core/services/feeds/connection_manager.go

```diff
@@ -159,7 +162,7 @@ func (mgr *connectionsManager) Close() {
 // GetClient returns a single client by id
 func (mgr *connectionsManager) GetClient(id int64) (pb.FeedsManagerClient, error) {
 	conn, ok := mgr.connections[id]
-	if !ok {
+	if !ok || !conn.connected {
```

Comment:
Not sure I follow how this helps since it gets set to true the same time we set connections[id] ?

Reply:
It's initially set to false on line 83. It only gets updated to true after the the initial Dial succeeds on line 114